src/charm.py

```diff
@@ -143,6 +144,38 @@ def snap_path(self) -> Optional[str]:
 
         return self._snap_path
 
+    @property
+    def snap_channel(self) -> str:
+        """Get the channel for exporter snap.
+
+        The channel is determined by the controller version that the charm is deployed
+        under. In case the controller version is less than 2.9, the snap channel is set
+        to 2.8/stable. Otherwise, the snap channel is set to 2.9/stable.
+        """
+        controller_version = self.get_controller_version()
+
+        if controller_version < version.parse("2.9"):
+            return "2.8/stable"
+        if (
+            controller_version.major == version.parse("2.9").major
+            and controller_version.minor == version.parse("2.9").minor
```

Comment:
```suggestion
            version.parse("2.9") <= controller_version < version.parse("3.0")
```

Reply:
This is no longer relevant with the new implementation. I'm resolving this conversation.